blueocean-jira/src/main/java/io/jenkins/blueocean/service/embedded/jira/BlueJiraIssue.java

```diff
@@ -97,9 +97,10 @@ public int hashCode() {
                 return null;
             }
             Collection<String> issueKeys = findIssueKeys(changeSetEntry.getMsg(), site.getIssuePattern());
-            Iterable<BlueIssue> transformed = Iterables.transform(issueKeys,
-                                                                  s -> BlueJiraIssue.create(site, action.getIssue(s)));
-            return ImmutableList.copyOf(Iterables.filter(transformed, Predicates.notNull()));
+            Collection<BlueIssue> blueIssues = issueKeys.stream().map(s -> BlueJiraIssue.create(site, action.getIssue(s)))
+                                                        .filter(Objects::nonNull)
+                                                        .collect(Collectors.toList());
```

Comment:
should be about the same and the test passes

Reply:
You can save one step, the `blueIssues` variable, doing it all together:
`.collect(collectingAndThen(toList(), ImmutableList::copyOf));`

Reply:
`ImmutableList` is from Guava?

Reply:
Oh shit, I copied the guava version. I meant `.collect(collectingAndThen(Collectors.toList(), Collections::unmodifiableCollection));`